olp-cpp-sdk-authentication/src/AuthenticationClientImpl.cpp

```diff
@@ -555,81 +555,46 @@ client::CancellationToken AuthenticationClientImpl::HandleUserRequest(
-  request.WithHeader(http::kContentTypeHeader, kApplicationJson);
-  request.WithHeader(http::kUserAgentHeader, http::kOlpSdkUserAgent);
-  request.WithSettings(std::move(network_settings));
+    client::OlpClient client = CreateOlpClient(settings_, {}, false);
 
-  std::shared_ptr<std::stringstream> payload =
-      std::make_shared<std::stringstream>();
-  request.WithBody(GenerateSignUpBody(properties));
-  auto send_outcome = settings_.network_request_handler->Send(
-      request, payload,
-      [callback, payload,
-       credentials](const http::NetworkResponse& network_response) {
-        auto response_status = network_response.GetStatus();
-        auto error_msg = network_response.GetError();
+    const auto url = settings_.token_endpoint_url + kUserEndpoint;
+    auto auth_header = GenerateAuthorizationHeader(
+        credentials, url, std::time(nullptr), GenerateUid());
 
-        if (response_status < 0) {
-          // Network error response
-          AuthenticationError error(response_status, error_msg);
-          callback(error);
-          return;
-        }
+    client::OlpClient::ParametersType headers = {
+        {http::kAuthorizationHeader, std::move(auth_header)}};
 
-        auto document = std::make_shared<rapidjson::Document>();
-        rapidjson::IStreamWrapper stream(*payload.get());
-        document->ParseStream(stream);
-
-        std::shared_ptr<SignUpResultImpl> resp_impl =
-            std::make_shared<SignUpResultImpl>(response_status, error_msg,
-                                               document);
-        SignUpResult response(resp_impl);
-        callback(response);
-      });
+    auto signup_response = client.CallApi(kUserEndpoint, "POST", {}, headers,
+                                          {}, GenerateSignUpBody(properties),
+                                          kApplicationJson, std::move(context));
 
-  if (!send_outcome.IsSuccessful()) {
-    thread::ExecuteOrSchedule(settings_.task_scheduler, [send_outcome,
-                                                         callback] {
-      std::string error_message =
-          ErrorCodeToString(send_outcome.GetErrorCode());
-      AuthenticationError result({static_cast<int>(send_outcome.GetErrorCode()),
-                                  std::move(error_message)});
-      callback(result);
-    });
-    return client::CancellationToken();
-  }
+    const auto status = signup_response.GetStatus();
+    std::string response_text;
+    signup_response.GetResponse(response_text);
+    if (status < 0) {
+      return client::ApiError(status, response_text);
+    }
 
-  auto request_id = send_outcome.GetRequestId();
-  return client::CancellationToken([weak_network, request_id]() {
-    auto network = weak_network.lock();
+    auto document = std::make_shared<rapidjson::Document>();
+    document->Parse(response_text.c_str());
+    return {std::make_shared<SignUpResultImpl>(
+        status, olp::http::HttpErrorToString(status), document)};
```

Comment:
`olp::http::HttpErrorToString(status)` is definitely not correct. Please have a look at `OlpClient/SendRequest` to know what is the transition between `NetworkResponse` and `HttpResponse`, and in which case you should have an error and in which case you should not.

Reply:
Checked a text message returned from GetResponse, it uses the same code but  internally in OlpClient/ExecuteSingleRequest.